jdk/test/jdk/internal/platform/cgroup/TestCgroupSubsystemFactory.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2020, Red Hat Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import jdk.internal.platform.CgroupSubsystemFactory;
+import jdk.internal.platform.CgroupSubsystemFactory.CgroupTypeResult;
+import jdk.testlibrary.Utils;
+import jdk.testlibrary.FileUtils;
+
+
+/*
+ * @test
+ * @requires os.family == "linux"
+ * @modules java.base/jdk.internal.platform
```

Comment:
this is benign but potentially misleading in 8u, should we remove it? I only hesitate because it might mean rebasing up to 8 pull requests on top

Reply:
Let's not fix this in this integration batch. If we really want to, we can remove those with a single 8u-only bug. FWIW, there are more of those `@modules` lines in tests elsewhere (only two of them are from container tests):

```
$ grep -rn '@modules' jdk/test/ | wc -l
77
```